distributed/protocol/compression.py

```diff
@@ -60,13 +60,28 @@ def _fixed_snappy_decompress(data):
         lz4_compress = lz4.LZ4_compress
         lz4_decompress = lz4.LZ4_uncompress
 
-    def _fixed_lz4_decompress(data):
-        # helper to bypass missing memoryview support in lz4
-        if isinstance(data, memoryview):
-            data = data.tobytes()
-        return lz4_decompress(data)
+    # helper to bypass missing memoryview support in current lz4
+    # (fixed in later versions)
+
+    def _fixed_lz4_compress(data):
+        try:
+            return lz4_compress(data)
+        except TypeError:
+            if isinstance(data, memoryview):
+                return lz4_compress(data.tobytes())
+            else:
+                raise
 
-    compressions['lz4'] = {'compress': lz4_compress,
+    def _fixed_lz4_decompress(data):
+        try:
+            return lz4_decompress(data)
+        except TypeError:
+            if isinstance(data, memoryview):
+                return lz4_decompress(data.tobytes())
+            else:
+                raise
```

Comment:
Is there a cost to this exception?  If we know an explicit version then it might be better to perform this check once:

```python
if lz4.__version__ > ...:
    _fixed_lz4_compress = lz4_compress
else:
    def _fixed_lz4_compress(data):
        ...
```